sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/LineaSendBundleTransactionSelector.java

```diff
@@ -0,0 +1,91 @@
+package net.consensys.linea.sequencer.txselection.selectors;
+
+import static java.lang.Boolean.TRUE;
+
+import java.util.List;
+import java.util.Optional;
+
+import net.consensys.linea.rpc.services.LineaLimitedBundlePool;
+import org.hyperledger.besu.datatypes.Hash;
+import org.hyperledger.besu.datatypes.PendingTransaction;
+import org.hyperledger.besu.plugin.data.TransactionProcessingResult;
+import org.hyperledger.besu.plugin.data.TransactionSelectionResult;
+import org.hyperledger.besu.plugin.services.txselection.PluginTransactionSelector;
+import org.hyperledger.besu.plugin.services.txselection.TransactionEvaluationContext;
+
+public class LineaSendBundleTransactionSelector implements PluginTransactionSelector {
+
+  public static final long DEFAULT_BUNDLE_POOL_SIZE = 10000L;
+  final LineaLimitedBundlePool bundlePool;
+
+  public LineaSendBundleTransactionSelector(LineaLimitedBundlePool bundlePool) {
+    this.bundlePool = bundlePool;
+  }
+
+  @Override
+  public TransactionSelectionResult evaluateTransactionPreProcessing(
+      final TransactionEvaluationContext<? extends PendingTransaction> txContext) {
+    final var blockHeader = txContext.getPendingBlockHeader();
+    // map the pending transaction to its origin bundle:
+    var bundle =
+        bundlePool.getBundleByPendingTransaction(
+            blockHeader.getNumber(), txContext.getPendingTransaction());
+
+    if (bundle.isPresent()) {
+      var satisfiesCriteria =
+          bundle.stream()
+              // filter if we have a min timestamp that has not been satisfied
+              .filter(
+                  b ->
+                      b.minTimestamp()
+                          .map(minTime -> minTime < System.currentTimeMillis())
+                          .orElse(TRUE))
+              // filter if we have a max timestamp that has not been satisfied
+              .filter(
+                  b ->
+                      b.maxTimestamp()
+                          .map(maxTime -> maxTime > System.currentTimeMillis())
+                          .orElse(TRUE))
+              .findAny();
+      if (satisfiesCriteria.isPresent()) {
+        return TransactionSelectionResult.SELECTED;
+      } else {
+        return TransactionSelectionResult.invalid("Failed Bundled Transaction Criteria");
+      }
+    }
+
+    // TODO: if we do not find the bundle in the pool for 'reasons' this is an incredibly weak yes.
+    //  It could have been evicted due to age or size, or it could be assocaited with a different
+    //  block number than the one we are building.I don't like it.
```

Comment:
Yes this is an open point, actually if I am not wrong we can make this selector optional, passing the required information (in this case the pending block header) to 
`public void selectPendingTransactions(
      final BlockTransactionSelectionService blockTransactionSelectionService)`
to only select from the private txpool the pending txs that are candidate for the current block, that will also avoid doing unneeded work to evaluate bundles that for sure will not be selected.
So probably we can move this filtering before, will extend the API to add `ProcessableBlockHeader` to the method

Reply:
I am reasonably happy with the short circuit for non bundle transactions and a small eval cache that should not have any evictions during bundle evaluation. 👍